### PR TITLE
Recommend

### PR DESCRIPTION
**Pull request by Chirag-Khatri-CK**

### CODE REVIEW

File: src/configurations/messages.js [hunk 1]
```diff
@@ -14,6 +14,10 @@ const messagingConfig = {
         notification: {
             watched_all_cases: 'You have watched all cases. Please wait till someone raises some.',
             failure: 'Failed to retrieve {0}. Please try again'
+        },
+        recommendation: {
+            watched_all_recommends: 'You have watched all recommends. Please wait till new recommends.',
+            failure: 'Failed to retrieve recommendation. Please try again'
```
Comment (chayan007) [inline src/configurations/messages.js:20]: Create a common error message for watched all:
```
watched_all:  "You have watched all {0}. Please wait for new {0}" [recommendations]
```
For failure, please use `unknown_error` for that
Also, delete `notification` and `recommendation` blocks.

File: src/components/recommendation/PeopleCard.vue [hunk 2]
```diff
@@ -1,58 +1,86 @@
...
+            a4
+            profile-image
+            bg-primary
+            shadow-inset
+            border border-light
+            rounded-circle
+          "
+        >
+          <img
+            :src="getImg()"
```
Comment (chayan007) [inline src/components/recommendation/PeopleCard.vue:16]: use `v-if` instead of calling functions.

Comment (chayan007) [reply src/components/recommendation/PeopleCard.vue:16]: check if `this.recommendation.display_pic` is present

File: src/configurations/user.js [hunk 3]
```diff
@@ -61,6 +61,9 @@ if (process.env.NODE_ENV === "PRODUCTION") {
             getProfile:{
                 "endpoint": "profile/user/{0}/"
             },
+            fetchRecommendation:{
```
Comment (chayan007) [inline src/configurations/user.js:64]: *`recommendations`

File: src/services/recommendation.service.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { stringFormat } from '@/helpers';
+import { config } from "@/configurations";
+import axios from "axios";
+
+export const recommendationService = { getRecommendations };
+
+function getRecommendations() {
```
Comment (chayan007) [inline src/services/recommendation.service.js:7]: To be moved into user service.

File: src/views/protected/Recommendation.vue [hunk 5]
```diff
@@ -1,32 +1,77 @@
...
+          this.recommendations = recommendations;
+        })
+        .catch(() => {
+          throw stringFormat(
+            config.messagingConfig.messages.recommendation.failure,
+            "recommendations"
+          );
+        });
+    },
+    open_case() {
```
Comment (chayan007) [inline src/views/protected/Recommendation.vue:65]: @Chirag-Khatri-CK what is the purpose of this ?

**Review by chayan007 — CHANGES_REQUESTED** [review]

File: src/services/recommendation.service.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    let url = `${config.commonConfig.$apiUrl}/${config.userConfig.api.fetchRecommendation.endpoint}`;
+
+    return axios
+    .get(url)
```
Comment (chayan007) [inline src/services/recommendation.service.js:11]: `return axios.get(url)` in same line

File: src/views/protected/Recommendation.vue (hunk 5 above)
Comment (Chirag-Khatri-CK) [reply src/views/protected/Recommendation.vue:65]: Sir actually I am rendering case.vue to check I make use of this.

File: src/services/recommendation.service.js (hunk 4 above)
Comment (Chirag-Khatri-CK) [reply src/services/recommendation.service.js:7]: Done Sir

File: src/services/recommendation.service.js (hunk 6 above)
Comment (Chirag-Khatri-CK) [reply src/services/recommendation.service.js:11]: Done Sir

File: src/configurations/user.js (hunk 3 above)
Comment (Chirag-Khatri-CK) [reply src/configurations/user.js:64]: Done Sir

File: src/configurations/messages.js [hunk 7]
```diff
@@ -10,11 +10,9 @@ const messagingConfig = {
             user_not_found: 'No records found for the username / email.',
             does_not_exist_error: 'This {0} does not exist. Please check again.',
             otp_not_match: 'Wrong OTP submitted. Please request new OTP and try again',
-        },
-        notification: {
             watched_all_cases: 'You have watched all cases. Please wait till someone raises some.',
-            failure: 'Failed to retrieve {0}. Please try again'
-        }
+            watched_all_recommends: 'You have watched all {0}. Please wait for {0}.'
```
Comment (chayan007) [inline src/configurations/messages.js:14]: `watched_all`

**Review by chayan007 — APPROVED** [review]

**Pull request merged**